armi/reactor/components/component.py

```diff
@@ -402,6 +407,11 @@ def adjustDensityForHeightExpansion(self, newHot):
         See AxialExpansionChanger.
         """
         self.changeNDensByFactor(1.0 / self.getHeightFactor(newHot))
+        if self.p.detailedNDens:
```

Comment:
This feels like this new code would be reduced throughout the code base if it was inside of `changeNDensByFactor`. Can that not be done?

Reply:
I agree with Jake. This should be able to be done. That would greatly simplify the PR to updated the one method. And will make required testing much simpler too.

Reply:
I agree with Jake. This should be able to be done. That would greatly simplify the PR to updated the one method. And will make required testing much simpler too.

Reply:
ok I moved the functionality into `changeNDensByFactor` and modified a test to cover it.